File: solutions/cpp/nth-prime/1/nth_prime.cpp
#include "nth_prime.h"
#include <cmath>
#include <stdexcept>


namespace nth_prime {

int nth(int N){
    if(N<1){
        throw std::domain_error("Number must be >=1.");
    } else {
        if (N==1) 
            return 2; 
        int count = 2;
        int candidate = 3;
        while(count != N){
           candidate += 2;
            bool is_prime = true;
            for(int i = 3; i<= round(sqrt(candidate));i+=2){
             if (!(candidate % i)) {
                  is_prime = false;
                  break;
              }               
             }  
           if (is_prime)
                 count++;
           }
        return candidate;
    }
}

}  // namespace nth_prime

File: solutions/fortran/nth-prime/1/nth_prime.f90
module nth_prime
  implicit none
contains
! get nth prime
  integer function prime(n)
    integer, intent(in) :: n
    integer :: p_no, cand, i
    logical :: is_prime
    prime = -1 
    if (n>0) then
      if (n==1) then
        prime = 2
      else
        p_no = 2
        cand = 3
        is_prime = .false.
        do while (p_no /= n) 
          cand = cand + 2
          is_prime = .true.
          do i = 3, floor(sqrt(real(cand+1))), 2
            if (mod (cand,i) == 0) then
              is_prime = .false.
              exit
            end if
          end do
          if (is_prime) then 
            p_no  = p_no + 1
          end if
        end do
        prime = cand
      end if
    end if
  end function


end module